src/components/Item/VisualizationItem/Item.js

```diff
@@ -96,6 +96,11 @@ class Item extends Component {
                 await apiFetchVisualization(this.props.item)
             )
 
+        // force fetch when recording to allow caching of the visualizations request
+        if (this.props.isRecording) {
+            apiFetchVisualization(this.props.item)
```

Comment:
There is no need to await this request?

It does look a bit weird to have the same request 5 lines above.